python/main.py

```diff
@@ -18,21 +23,54 @@
     allow_headers=["*"],
 )
 
+#items_json = pathlib.Path(__file__).parent.resolve() / "items.json"
+
+#モード
+#r:読み込み（デフォルト）
+#w:書き込み
+#a:追記
+#r+:読み込みと書き込み
 
 @app.get("/")
-def root():
+async def root():
     return {"message": "Hello, world!"}
 
-
 @app.post("/items")
-def add_item(name: str = Form(...)):
-    logger.info(f"Receive item: {name}")
-    return {"message": f"item received: {name}"}
+async def add_item(name: str = Form(...), category: str=Form(...), image: UploadFile= File(...)):
+    logger.info(f"Receive item: {name},{category},image:{image.filename}")
+    
+    images=pathlib.Path(__file__).parent/"python"/"images"
+    #アップロードされたファイルの内容を非同期で読み込む
+    contents= await image.read()
+    hash_sha256=hashlib.sha256(contents).hexdigest()
+    image_filename=f"{hash_sha256}.jpg"
+    image_path=images/image_filename
+    with open(image_path,"wb") as f:
+        f.write(contents)
+        
+    #loadでjsonファイルをPythonのデータ構造に変換する
+    with open("items.json","r") as f:
+        json_load=json.load(f)
+    
+    new_item={"items": [{"name": name, "category": category,"image_name": image_filename}]}
+    json_load.append(new_item)
+    #書き込みモードにしてnew_itemをitems.jsonに追加する
+    with open("items.json","w") as f:
+        json_dump=json.dump(json_load, f,indent=4)
+    
+
+    return {"message": f"item received: {name},{category},{image}"}
+
+@app.get("/items")
+def get_items():
+    return json_load
 
+@app.get("/items/{items_id}")
+def get_items(items_id:int):
```

Comment:
This function is missing a body. It also has the same name as the previous function (`get_items`)

Reply:
Thank you